pkg/shares/parse_compact_shares.go

```diff
@@ -61,7 +51,7 @@ func parseRawData(rawData []byte) (units [][]byte, err error) {
 		if err != nil {
 			return nil, err
 		}
-		if unitLen == 0 {
+		if unitLen == 0 || unitLen > uint64(len(actualData)) {
```

Comment:
[question] should `unitLen > uint64(len(actualData))` return an error?

Reply:
added a comment explaining it